### PR TITLE
Iss 205

### LINKED ISSUE #205: Tune EKF For OSV Platform
Tuning covariance matrix for the EKF with the most recent bags.

### PR DESCRIPTION
**Pull request by joeyzhu00**
Closes #205 

Overall, the EKF seems to be fairly robust in many cases except for edge cases such as during some corners (possibly due to the a non-level ground) and when the ETH-Zurich package does not filter the output when the GPS signals start reflecting off the windows at the Student Union. One change that we can implement in a future issue is to ignore GPS signals when we're within a certain X-Y position on campus.

### CODE REVIEW

File: src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp [hunk 1]
```diff
@@ -150,13 +182,37 @@ namespace frame_calibration_node
                 geodesy_tf_msg.pose.pose.position.x = geodesy_tf_point.point.x;
                 geodesy_tf_msg.pose.pose.position.y = geodesy_tf_point.point.y;
                 geodesy_tf_msg.header.frame_id = geodesy_tf_point.header.frame_id;
+                // geodesy_tf_msg.child_frame_id = temp_geodesy_tf_point.header.frame_id;
                 geodesy_tf_msg.header.stamp = ros::Time::now();
 
-                geodesy_tf_pub.publish(geodesy_tf_msg);
+                // counter to remove edge case of first data point
```
Comment (drofp) [inline src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp]: If this is its only purpose, maybe consider replacing with boolean for clarity and to save memory

File: src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp [hunk 2]
```diff
@@ -150,13 +182,37 @@ namespace frame_calibration_node
...
+                
+                if(gps_msg_counter == 1)
+                {
+                    geodesy_tf_pub.publish(geodesy_tf_msg);    
+                }
+                else
+                {
+                    // calculate the euclidian distance between point (n) and point (n-1)
+                    float distance_comparison;
+                    float gps_threshold = 0.5; // [m]
```
Comment (drofp) [inline src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp]: Consider using a rosparam for this, so it's easily configurable from a launch file

File: src/modules/localization/src/wheel_odom/odometry.cpp [hunk 3]
```diff
@@ -3,9 +3,7 @@
 /*
 Takes wheel odometry from Arduino encoders and translates to vehicle velocity and position
 NOTE: This script is to handle the raw wheel odometry values from the Signwise 600 P/R rotary encoder.
-      It also takes in various inputs from topics yaw from the IMU to dead-reckon position and velocity in the X and Y frame.
-      Dead-reckoning requires previous pose to be tracked, we take advantage of this by subscribing to
-      the output of the EKF and using those pose messages as the previous pose of the wheel odometry.
+      It also takes in various inputs from topics yaw from the IMU to calculate velocity in the X and Y frame.
```
Comment (drofp) [inline src/modules/localization/src/wheel_odom/odometry.cpp]: It looks like IMU subscription has been commented out. Is this documentation still valid?

File: src/modules/localization/src/wheel_odom/odometry.cpp [hunk 4]
```diff
@@ -129,53 +127,55 @@ namespace wheel_odometry_node
...
-        y_velocity = vel_magnitude * sin(yaw_estimate); // [m/s]
-
+        // x_velocity = vel_magnitude * cos(yaw_estimate); // [m/s]
+        // y_velocity = vel_magnitude * sin(yaw_estimate); // [m/s]
+        x_velocity = vel_magnitude;
+        // y_velocity = 0;
         // twist messages
         velocity_estimate.twist.linear.x = x_velocity; // [m/s]
-        velocity_estimate.twist.linear.y = y_velocity; // [m/s]
+        // velocity_estimate.twist.linear.y = y_velocity; // [m/s]
```
Comment (drofp) [inline src/modules/localization/src/wheel_odom/odometry.cpp]: When would we use `y_velocity` in the future?

Comment (joeyzhu00) [reply src/modules/localization/src/wheel_odom/odometry.cpp]: It's unlikely, I can delete the excess within `odometry.cpp`

File: src/modules/localization/src/wheel_odom/odometry.cpp (hunk 3 above)
Comment (joeyzhu00) [reply src/modules/localization/src/wheel_odom/odometry.cpp]: I'll edit the documentation and delete the excess comments, the transform will just be done within the `robot_localization` package.

File: src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp (hunk 2 above)
Comment (joeyzhu00) [reply src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp]: Ok

**Review by drofp — APPROVED** [review]

**Comment by menson168** [comment]
Looks good! Approved.

**Pull request merged**